pkg/operator/resource/resourceapply/generic.go

```diff
@@ -34,7 +34,9 @@ type ApplyResult struct {
 	Error   error
 }
 
-func ApplyDirectly(client kubernetes.Interface, recorder events.Recorder, manifests AssetFunc, files ...string) []ApplyResult {
+// ApplyDirectly applies the given manifest files to API server.
+// NOTE: Do not pass CR to this function as it will break if the client is using protobuf.
```

Comment:
isn't the switch excluding them anyway?

Reply:
@sttts wrong wording, I wanted to discourage people to add CR inside that switch.